subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -396,41 +489,82 @@ private void parseData(String data) {
       }
     } catch (Exception e) {
       System.err.println("Exception parseData:" + e.getMessage());
-      System.exit(0);
     }
   }
 
+  private HashMap dataToMap(String[] expected_key, String[] data_array) {
+    HashMap<String, String> hashMap = new HashMap<String, String>();
+    for (int i = 0; i < expected_key.length; i++) {
+      hashMap.put(expected_key[i], data_array[i]);
+    }
+    return hashMap;
+  }
+
   private void parseRequestFlag(String data, int requestFlag) {
-    switch (requestFlag) {
-      case 2:
-        // parse show interface
-        login_report += "show interface:\n";
-        parse_packet(data, show_interface_port_expected, show_interface_port_pointers);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 3:
-        // parse show platform
-        login_report += "\nshow platform:\n";
-        parse_packet(data, show_platform_port_expected, show_platform_port_pointers);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 4:
-        // parse show run
-        data = cleanShowRunData(data);
-        login_report += "\n" + data;
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 5:
-        // parse show stack
-        login_report += "\nshow stack:\n";
-        parse_inline(data, show_stack_expected, show_stack_pointers, show_stack_data);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
+    try {
+      switch (requestFlag) {
+        case 2:
+          // parse show interface
+          login_report += "show interface:\n";
+          parse_packet(data, show_interface_port_expected, show_interface_port_pointers);
+          interface_map = dataToMap(interface_expected, show_interface_data);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 3:
+          // parse show platform
+          login_report += "\nshow platform:\n";
+          parse_packet(data, show_platform_port_expected, show_platform_port_pointers);
+          platform_map = dataToMap(platform_expected, show_platform_data);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 4:
+          // parse show power-inline
+          login_report += "\nshow power-inline:\n";
+          if (data.contains("Power-inline is disabled")) {
```

Comment:
@ewRedstone Whether the PoE test is run now depends on:
a) telnet being enabled on the switch
b) `show power-inline` being an available command on the switch
c) The return string from `show power-inline` returning 'Power-inline is disabled'

I'm not sure if these three conditions will always be met and, if they're not, the operator might get expect the PoE test to run but not see results in the final report. The test might fail/hang unexpectedly. As long as this is documented clearly in the PoE test's README.md this should be fine but:

Having a `poe` object in module_config.json as in:

```
"switch": {
     "enabled": true,
     "poe": {
       "enabled": "false"
     }
``` 

a) Allows enabling/disabling of other PoE tests in the future (e.g. support/power/negotiation, which would be other keys in the `poe` object) by the operator
b) Decouples the result of the test from the availabilty of telnet
c) Decoupled the result of the test from the exact sytax of the Allied Telesis switch